source/simulator/core.cc

```diff
@@ -1976,6 +1974,23 @@ namespace aspect
           break;
         }
 
+        case NonlinearSolver::Binary_input:
+        {
+          AssertThrow (!parameters.free_surface_enabled,
+                       ExcMessage (std::string("Currently, we do not support free surface mesh with the \"nonlinear\" binary input scheme.")));
+
+          std::string filename = parameters.binary_data_directory + "/" + parameters.binary_data_file_name + Utilities::int_to_string(timestep_number, 5) + ".mesh";
+
+          triangulation.load(filename.c_str());
```

Comment:
Does this work? You are replacing the current mesh by a different one. We need to figure out that the mesh is identical right?